bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/handler/RuleEnabledHandler.java

```diff
@@ -0,0 +1,160 @@
+     */
+    private String[] rulesIds;
+
+    /**
+     * This field will store the action that should be applied.
+     */
+    private boolean enabled;
+
+    /**
+     * This Tracker is used to manage ruleAdmin field.
+     */
+    private ServiceTracker tracker = null;
+
+    /**
+     * This field will be used to manage the rules.
+     */
+    private RuleRegistry registry = null;
+
+    /**
+     * Reference to the bundle context of the framework.
+     */
+    private BundleContext context = null;
+
+    /**
+     * This field will be true if at some point in execution of this {@link Action} the {@link RuleRegistry} becomes
+     * unavailable.
+     */
+    private boolean isInterrupted = false;
+
+    @SuppressWarnings({ "unchecked" })
+    public RuleEnabledHandler(Action module, BundleContext context) {
+        super(module);
+        final Configuration config = module.getConfiguration();
+        if (config == null) {
+            throw new IllegalArgumentException("'Configuration' can not be null.");
+        }
+
+        String action = (String) config.get(ACTION_KEY);
+        if (action == null) {
+            throw new IllegalArgumentException("'action' property can't be undefined.");
+        }
+
+        if (action.equalsIgnoreCase("enable")) {
+            enabled = true;
+        } else if (action.equalsIgnoreCase("disable")) {
+            enabled = false;
+        } else {
+            throw new IllegalArgumentException(
+                    "permitted values for the 'action' property are 'enable' and 'disable'.");
+        }
+
+        final String rules = (String) config.get(RULE_UIDS_KEY);
+
+        if (rules == null) {
+            throw new IllegalArgumentException("'ruleIds' property can't be undefined.");
+        }
+        rulesIds = rules.split(",");
+
+        this.context = context;
+
+        try {
+            Filter filter = FrameworkUtil.createFilter(FILTER);
+            tracker = new ServiceTracker(context, filter, this);
+        } catch (InvalidSyntaxException e) {
+            // Should never happen.
+        }
+    }
+
+    @Override
+    public Map<String, Object> execute(Map<String, ?> context) {
+
+        try {
+            for (String entity : rulesIds) {
+                registry.setEnabled(entity, enabled);
+            }
+        } catch (NullPointerException e) {
+            isInterrupted = true;
+        }
+
+        isInterrupted = false;
+        return null;
+    }
+
+    @Override
+    public void dispose() {
+        if (tracker != null) {
+            tracker.close();
+            tracker = null;
+        }
+
+        super.dispose();
+    }
+
+    public void activate() {
+        tracker.open();
+    }
+
+    @SuppressWarnings({ "unchecked" })
+    @Override
+    public Object addingService(ServiceReference reference) {
+        registry = (RuleRegistry) context.getService(reference);
+        if (isInterrupted) {
+            execute(null);
```

Comment:
IMHO the tracker just has to track RuleRegistry service not to execute the command.